### PR TITLE
Feature/__main__

### PR DESCRIPTION
**Pull request by forslund**
## Description
Make the processes runable modules. This includes

- Replacing the `main.py` script with [`__main__.py`](https://docs.python.org/3.4/library/__main__.html).
- update logic in `start-mycroft.sh` and `stop-mycroft.sh` to handle this. This changes some of the logic to use [pgrep](https://linux.die.net/man/1/pgrep)
- Slight refactoring separating the bigger parts in the main-scripts into separate files leaving the `__main__.py` files caring mostly of the startup/setup logic.

## How to test
make sure start-mycroft.sh and stop-mycroft.sh bahaves as expected.

## Contributor license agreement signed?
CLA [Yes]

### CODE REVIEW

File: mycroft/client/text/__main__.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+sys.excepthook = custom_except_hook
+
+sys.stdout = io.StringIO()
+sys.stderr = io.StringIO()
+
+bSimple = '--simple' in sys.argv
+
+
+def main():
```
[the remark this answers is not in the document]
Comment (forslund) [reply mycroft/client/text/__main__.py]: Thanks, I'll fix that when I resolve all the conflicts

**Comment by MatthewScholefield** [comment]
I'm getting a bad magic number error that causes certain **skills** not to load (yet the skills process as a whole loads fine):
```
return load_source(name, filename, file)
File "/usr/lib/python3.6/imp.py", line 172, in load_source
module = _load(spec)
File "<frozen importlib._bootstrap>", line 684, in _load
File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 678, in exec_module
File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
File "/opt/mycroft/skills/mycroft-installer.mycroftai/__init__.py", line 22, in <module>
from mycroft.skills.main import SkillManager
ImportError: bad magic number in 'mycroft.skills.main': b'\x03\xf3\r\n'
```

Any idea why or what the best solution is? I'm guessing clearing my `pyc` files would fix it, but I'm  curious if it's only for my setup or will happen for others too.

**Comment by forslund** [comment]
In the case of the installer skill it's that `mycroft.skills.main` doesn't exist anymore so the skill can't load. The magic number thing is likely that a copy exists in the `__pycache__` which gets loaded, maybe...

**Comment by forslund** [comment]
Just checked my log, I did not get any bad magic number error just a ModuleNotFound error. Have you run Mycroft from a different python interpreter recently?

**Pull request merged**